### PR TITLE
Bug fixes to User.update() & documented query methods IDs

### PR DESCRIPTION
**Pull request by TomY-Zhang**

### CODE REVIEW

**Review by leo4life2 — DISMISSED** [review]

File: server/src/types/address.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def insert(data: dict) -> None:
 
         try:
             if Address.exists({'aid': data['aid']}):
-                raise errors.DuplicateKeyError()
+                raise errors.DuplicateKeyError(f'Address with address id {data["aid"]} already exists')
```
Comment (Olivernyu) [inline server/src/types/address.py]: Why are we raising an error? If two separate users try to insert the same address then the later one will see an error. Same comment for other two types. For login specifically, if someone tries to do login twice in a role, do they see an error? 
I think there are several approaches we can take, either do checks before insert() is called, or modify this function to do nothing when an entry already exists. Either way, I think the schema here needs to be more explicitly considered and documented so we don't have bugs later on.

**Review by Olivernyu — CHANGES_REQUESTED** [review]
I think we need to think more carefully on what to do when an insert happens with an already existing entry in our db. Raising an error for what the user will perceive and expect as a normal interaction does not seem appropriate.

File: server/src/types/address.py (hunk 1 above)
Comment (TomY-Zhang) [reply server/src/types/address.py]: I raised an error simply to log the error, but I agree with your point. I'll make changes to remove the error raising and include the update() function.

This week, I'll add documentation to the query methods as well as auto-incrementing IDs when inserting.

**Review by Olivernyu — APPROVED** [review]
lgtm

**Pull request merged**